go/ql/lib/semmle/go/security/LogInjectionCustomizations.qll

```diff
@@ -59,6 +59,77 @@ module LogInjection {
     }
   }
 
+  /**
+   * A call to `strings.NewReplacer`.
+   */
+  class StringsNewReplacerCall extends DataFlow::CallNode {
+    StringsNewReplacerCall() { this.getTarget().hasQualifiedName("strings", "NewReplacer") }
+
+    /**
+     * Gets an argument to this call corresponding to a string that will be
+     * replaced.
+     */
+    DataFlow::Node getAReplacedArgument() {
+      exists(int n | n % 2 = 0 and result = this.getArgument(n))
+    }
+  }
+
+  /**
+   * A configuration for tracking flow from a call to `strings.NewReplacer` to
+   * the receiver of a call to `strings.Replacer.Replace` or
+   * `strings.Replacer.WriteString`.
+   */
+  class StringsNewReplacerConfiguration extends DataFlow2::Configuration {
+    StringsNewReplacerConfiguration() { this = "StringsNewReplacerConfiguration" }
+
+    override predicate isSource(DataFlow::Node source) { source instanceof StringsNewReplacerCall }
+
+    override predicate isSink(DataFlow::Node sink) {
+      exists(DataFlow::MethodCallNode call |
+        sink = call.getReceiver() and
+        call.getTarget().hasQualifiedName("strings", "Replacer", ["Replace", "WriteString"])
+      )
+    }
+  }
```

Comment:
This part seems to be identical between `LogInjectionCustomizations` and `StringBreakCustomizations`. Would it make sense / be possible to de-duplicate this in a shared location?

Reply:
Good idea. It feels like something that should be usable in many different places. I've moved it to `StringOps.qll`, which should hopefully make that simple. I've started a performance evaluation as this is a fairly major change.

Reply:
The performance analysis was neutral.